packages/binding-mqtt/src/mqtt.ts

```diff
@@ -47,5 +47,23 @@ export class MqttForm extends Form {
 }
 
 export interface MqttClientConfig {
+    // username & password are redundated here (also find them in MqttClientSecurityParameters)
+    // because MqttClient.setSecurity() method can inject authentication credentials into this interface
+    // which will be then passed to mqtt.connect() once for all
+    username?: string;
+    password?: string;
     rejectUnauthorized?: boolean;
 }
+
+export interface MqttBrokerServerConfig {
+    uri: string;
+    user?: string;
+    psw?: string;
+    clientId?: string;
+    protocolVersion?: number;
```

Comment:
nitpicking: 3.1.1 is the actual version and I have no idea if there is a 3.0 or 3.1.0

Reply:
I've checked mqtt.js, AFIK they are just using the major protocol versions. I've changed the type to only accept 3, 4, or 5. We might need to think more how to handle this in the TD.

Reply:
Something really weird is that there is no MQTT spec with version 4, but [mqtt.js even defaults to that version](https://github.com/mqttjs/MQTT.js/blob/7ec4b8fd602e220f50693cb83f082dab764ed3f2/lib/client.js#L28) :D Not sure what to do here since a node-wot user does need to know about mqtt.js

Reply:
My proposal is to track this missaligment in an issue. Then we can investigate more and understand how we can map mqtt protolVersion onto mqtt.js versions.